benchmarks/work_precision_sets/time_vs_abstols.py

```diff
@@ -0,0 +1,115 @@
+import pybamm
+import numpy as np
+import matplotlib.pyplot as plt
+import itertools
+
+
+parameters = [
+    "Marquis2019",
+    "Prada2013",
+    "Ramadass2004",
+    "Chen2020",
+]
+
+models = {"SPM": pybamm.lithium_ion.SPM(), "DFN": pybamm.lithium_ion.DFN()}
+
+abstols = [
+    0.001,
+    0.0001,
+    1.0e-5,
+    1.0e-6,
+    1.0e-7,
+    1.0e-8,
+    1.0e-9,
+    1.0e-10,
+    1.0e-11,
+    1.0e-12,
+    1.0e-13,
+]
+
+solvers = {
+    "IDAKLUSolver": pybamm.IDAKLUSolver(),
+    "Casadi - safe": pybamm.CasadiSolver(),
+    "Casadi - fast": pybamm.CasadiSolver(mode="fast"),
+}
+
+
+fig, axs = plt.subplots(len(solvers), len(models), figsize=(8, 5))
+
+for ax, i, j in zip(
+    axs.ravel(),
+    itertools.product(models.values(), solvers.values()),
+    itertools.product(models, solvers),
+):
+
+    for params in parameters:
+        time_points = []
+        solver = i[1]
+
+        model = i[0].new_copy()
+        c_rate = 10
```

Comment:
When I use any `c_rate` = 1, I get solver errors for a lot of combinations -
```
traceback (most recent call last):
  File "d:\code\PyBaMM\benchmarks\work_precision_sets\time_vs_abstols.py", line 86, in <module>
    solver.solve(model, t_eval=t_eval)
  File "C:\Users\Vaibhav Chopra\AppData\Local\Programs\Python\Python39\lib\site-packages\pybamm\solvers\base_solver.py", line 1083, in solve        
    new_solution = self._integrate(
  File "C:\Users\Vaibhav Chopra\AppData\Local\Programs\Python\Python39\lib\site-packages\pybamm\solvers\casadi_solver.py", line 177, in _integrate  
    solution = self._run_integrator(
  File "C:\Users\Vaibhav Chopra\AppData\Local\Programs\Python\Python39\lib\site-packages\pybamm\solvers\casadi_solver.py", line 741, in _run_integrator
    raise pybamm.SolverError(e.args[0])
pybamm.expression_tree.exceptions.SolverError: .../casadi/interfaces/sundials/cvodes_interface.cpp:392: CVode returned "CV_RHSFUNC_FAIL". Consult CVODES documentation.
```
If I use a try-except block to skip through the error, I get different lengths of x and y coordinate lists which cannot be plotted together.
Should I manually create x and y coordinates while iterating through the combinations rather than using the list defined at the top directly?

Reply:
That's weird. Do you know exactly which combinations are actually failing?

Can you try setting `tmax = 3500` (like the example below) and see if that works better?

Reply:
All work now except `Prada2013` with Casadi - fast, so I replaced it with Ecker2015 which was previously giving error, but now works!